### DISCUSSION TITLE
API requests with session id?

### DISCUSSION THREAD

**Discussion by GamePlayer-8** [mudler/LocalAI#1298] [Q&A] [answered]
Good evening,

Is there an option to request a session token / session id of a conversation and use it for remembering the history of questions/answers for the AI model?

By default I can do:
```shell
curl http://localhost:8080/v1/completions -H "Content-Type: application/json" -d '{
     "model": "your-model.bin",
     "prompt": "A long time ago in a galaxy far, far away",
     "temperature": 0.7
   }'
```

although I can't find an info in the documentation about preserving a conversation history for the AI model till it's being deleted by the last POST request.

**Comment by lunamidori5** [comment]
This is a backend for apps that have that, so check with the openai v0 / openai v1 app that uses localai for memory, check out autogen / memgpt / a few others for more info <3

**Reply by lunamidori5** [reply, marked as answer]
Also, two things, 1 Completions are out of date on Openai V1, please use Chat API for that.
2 Running a model raw is not recommended see - [How to set up a model](https://localai.io/howtos/easy-model-import-downloaded/) -  for more info on setting up a model!

**Reply by GamePlayer-8** [reply]
I've used [Easy Setup - Curl](https://localai.io/howtos/easy-request-curl/) documentation.
Thank you for the explanation. ❤️
